### PR TITLE
Added logic for different keycloak auth paths

### PR DESCRIPTION
**Pull request by wcater803**
In the itless environment, the Keycloak authentication instance was upgraded.  In this upgrade the "/auth" endpoint was deprecated.  This logic keeps the existing logic with the hardcoded "login/auth/keycloak" endpoint, but also adds in the availability to overwrite that by using the keycloakAuthPath (in the frontend) and KEYCLOAK_AUTH_PATH (foir backend) environment variables.

### CODE REVIEW

File: frontend/src/services/keycloak.js [hunk 1]
```diff
@@ -6,6 +6,6 @@ export class KeycloakService {
   static login(url, realm, client_id) {
     const keycloakInstance = new Keycloak({url: url, realm: realm, clientId: client_id});
     keycloakInstance.init({onLoad: 'login-required', checkLoginIframe: false, responseMode: 'query',
-                           redirectUri: Settings.serverUrl + '/login/auth/keycloak'});
```
Comment (mshriver) [inline frontend/src/services/keycloak.js:9]: Since the server URL is the same both times, you can flip this logic a bit.

`redirectUri: Settings.serverUrl + Settings.keycloakAuthPath ? Settings.keycloakAuthPath : '/login/auth/keycloak'`

You may be able to use `||` to be even more direct, but I'd have to test it when the envvar is empty.

Comment (wcater803) [reply frontend/src/services/keycloak.js:9]: I modified the code to use the or operator, as long as the keycloakAuthPath is a [Falsey](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value it works fine

File: backend/ibutsu_server/util/keycloak.py [hunk 2]
```diff
@@ -32,6 +32,8 @@ def get_keycloak_config(is_private=False):
         config["icon"] = current_app.config["KEYCLOAK_ICON"]
     if current_app.config.get("KEYCLOAK_NAME"):
         config["display_name"] = current_app.config["KEYCLOAK_NAME"]
+    if current_app.config.get("KEYCLOAK_AUTH_PATH"):
```
[the remark this answers is not in the document]
Comment (wcater803) [reply backend/ibutsu_server/util/keycloak.py]: Modified to use more direct logic

**Review by mshriver — APPROVED** [review]
